examples/request_no_blocking_main_thread/pokemon_client.rs

```diff
@@ -0,0 +1,41 @@
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug)]
+pub struct Pokemon {
+    pub name: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ResultBody {
+    results: Vec<Pokemon>,
+}
+
+#[derive(Debug)]
+pub struct PokemonClient {
+    page: u32,
+}
+
+impl PokemonClient {
+    const URI: &'static str = "https://pokeapi.co/api/v2/pokemon";
+    const PAGE_LIMIT: &'static str = "100";
+
+    pub fn new() -> Self {
+        Self { page: 0 }
+    }
+
+    pub async fn get_pokemon_list(&mut self) -> Result<Vec<Pokemon>, reqwest::Error> {
+        let current_offset = self.page.clone().to_string();
```

Comment:
```suggestion
        let current_offset = self.page.to_string();
```
you don't need to clone a copy type